pkg/provisioner/provisioner/provisioner.go

```diff
@@ -0,0 +1,382 @@
+		// TODO: Replace stdout to log out
+		fmt.Println("[identifyProvisionerState] ceph bootstrap check is failed")
+		return CephadmPkgInstalled, err
+	}
+
+	// Confirm some Ceph health is returned
+	const cephHealthPrefix = "HEALTH_"
+	if !strings.Contains(outputBootstrap.String(), cephHealthPrefix) {
+		// TODO: Replace stdout to log out
+		fmt.Println("[identifyProvisionerState] ceph status does not return HEALTH_*")
+
+		// TODO: Make own error pkg of hypersds-provisioner
+		return CephadmPkgInstalled,
+			errors.New("Error on Ceph bootstrap, cech status result: \n" +
+				outputBootstrap.String())
+	}
+
+	// Check Ceph bootstrap is committed
+	_, _, committed, err := p.checkKubeObjectUpdated()
+	if err != nil {
+		// Other error occurred on checkKubeObjectUpdated
+		// TODO: Replace stdout to log out
+		fmt.Println("[identifyProvisionerState] k8s configmap and secret check is failed")
+		return CephBootstrapped, err
+	}
+
+	if !committed {
+		// TODO: Replace stdout to log out
+		fmt.Println("[identifyProvisionerState] k8s configmap and secret are not updated")
+		return CephBootstrapped, nil
+	}
+
+	return CephBootstrapCommitted, nil
+}
+
+// TODO: Replace config const to inputs (e.g. K8sConfigMap, etc)
+func (p *Provisioner) checkKubeObjectUpdated() (*corev1.ConfigMap, *corev1.Secret, bool, error) {
+	// Check ceph.conf is updated to ConfigMap
+	configMap := &corev1.ConfigMap{}
+	if err := p.clientSet.Get(context.TODO(), types.NamespacedName{Namespace: p.cephNamespace, Name: p.cephName}, configMap); err != nil {
+		// Configmap must exist
+		if kubeerrors.IsNotFound(err) {
+			// TODO: Replace stdout to log out
+			fmt.Println("ConfigMap must exist")
+			return nil, nil, false, nil
+		}
+		return nil, nil, false, err
+	}
+
+	// Bootstrap commit has not occurred
+	if configMap.Data == nil {
+		return nil, nil, false, nil
+	}
+
+	// Check client.admin.keyring is updated to Secret
+	secret := &corev1.Secret{}
+	if err := p.clientSet.Get(context.TODO(), types.NamespacedName{Namespace: p.cephNamespace, Name: p.cephName}, secret); err != nil {
+		if kubeerrors.IsNotFound(err) {
+			// TODO: Replace stdout to log out
+			fmt.Println("Secret must exist")
+			return nil, nil, false, nil
+		}
+		return nil, nil, false, err
+	}
+
+	if secret.Data == nil {
+		return nil, nil, false, nil
+	}
+
+	return configMap, secret, true, nil
+}
+
+// NewProvisioner creates Provisioner using ceph deployment information and checks the current ceph deployment status in node
+func NewProvisioner(os wrapper.OsInterface, cephClusterSpec hypersdsv1alpha1.CephClusterSpec, clientSet client.Client, cephNamespace, cephName string) *Provisioner {
+	provisionerInstance := &Provisioner{}
+
+	// setCephCluster is only called once, on init
+	// No one is allowed to modify CephCluster
+	err := provisionerInstance.setCephCluster(cephClusterSpec)
+	if err != nil {
+		// TODO: Replace stdout to log out
+		fmt.Println("[Provisioner] setCephCluster Error")
+		panic(err)
```

Comment:
Provisioner package should not panic operator. Operator should decide whether it should call erroneous exit.

I recommend
- Change `NewProvisioner` function to return Provisioner and error
- Return error instead of calling panic